app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -35,13 +36,18 @@
 
     private final HabitList _habitList = new HabitList();
     private HabitItemAdapter _habitItemAdapter;
-    FirebaseFirestore _db;
+    FirebaseFirestore _db = FirebaseFirestore.getInstance();
 
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater,
                              ViewGroup container, Bundle savedInstanceState) {
         Context context = getContext();
-        _habitItemAdapter = new HabitItemAdapter(context, _habitList);
+
+        // populate the list with existing items in the database
+        FirestoreRecyclerOptions<Habit> options = new FirestoreRecyclerOptions.Builder<Habit>()
+                .setQuery(_db.collection("Habits"), Habit.class)
+                .build();
+        _habitItemAdapter = new HabitItemAdapter(options);
```

Comment:
Using some built-in methods provided by `Firebase`, it's easy to build up a `RecyclerView` list as shown here. You need to provide the `query`, along with the class it maps to, and pass it into your adapter.